.github/workflows/e2etests.yaml

```diff
@@ -0,0 +1,127 @@
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Get version of adapter
+        run: |
+          if [ ${{ github.event_name }} == "release" ];then
+              echo "version=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+          else 
+              echo "version=edge" >> $GITHUB_ENV
+          fi    
+      - name: Get latest release tag 
+        id: gettag  
+        uses: pozetroninc/github-action-get-latest-release@master
+        with:
+          repository: hashicorp/consul-k8s
+          excludes: prerelease, draft, edge  
+      - name: Change service mesh version in patternfile
+        run: | 
+           yq e -i '.services.consul.settings.version="${{ steps.gettag.outputs.release }}"' ./.github/install/deploy.yaml 
+           cat ./.github/install/deploy.yaml  
+      - name: Uploading file
+        uses: actions/upload-artifact@v2
+        with:
+          name: patternfile
+          path: ./.github/install/deploy.yaml
+
+  TestAdapterForServicemeshInstall:
+    needs: SetPatternfileForServicemeshInstall
+    uses: meshery/meshery/.github/workflows/test_adapters.yaml@master   
+    with: 
+      expected_resources: consul-client, consul-server
+      expected_resources_types: pod,pod
+      expected_resources_namespaces: consul, consul
+      deployment_url: https://raw.githubusercontent.com/meshery/meshery/master/install/deployment_yamls/k8s/meshery-consul-deployment.yaml
+      service_url: https://raw.githubusercontent.com/meshery/meshery/master/install/deployment_yamls/k8s/meshery-consul-service.yaml
+      adapter_name: consul
+      patternfile_name: deploy.yaml 
+      provider: Local 
+      sm_version: ${{ needs.SetPatternfileForServicemeshInstall.outputs.sm_version }}
+      adapter_version: ${{ needs.SetPatternfileForServicemeshInstall.outputs.adapter_version }}
+      output_filename: data.json
+    secrets:
+      token: ${{ secrets.PROVIDER_TOKEN }}  
+
+  UpdateDocsForServicemeshInstall:
+    needs: TestAdapterForServicemeshInstall
+    if:  "always() && github.event_name != 'pull_request' "
+    runs-on: ubuntu-latest
+    steps:
+      - name: Skip if needed
+        run: |
+            echo "this is it ${{ needs.SkipIfNeeded.outputs.skipdocupdate }} "
+            if [ "${{github.event_name }}" == "push" ];then
+              echo "version=edge" >> $GITHUB_ENV
+            fi
+            if [ "${{github.event_name }}" == "release" ];then
+              echo "version=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+            fi
+      - uses: actions/checkout@master
+        with:
+          repository: meshery/meshery 
+          token: ${{ secrets.GH_ACCESS_TOKEN }}   
+      - name: DownloadJSON
+        uses: actions/download-artifact@v2
+        with:
+          name: data.json
+      - name: Add commit SHORT_SHA
+        run: echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
+      - name: echo results
+        run: |
```

Comment:
@asubedy is this going to work for the integration tests / compatibility matrix in Meshery Docs?